src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -827,17 +827,18 @@ public INamedTypeSymbol GetTypeByMetadataName(string fullyQualifiedMetadataName)
         protected abstract INamedTypeSymbol CommonGetTypeByMetadataName(string metadataName);
 
         /// <summary>
-        /// Returns a new INamedTypeSymbol with the given element types and (optional) element names.
+        /// Returns a new INamedTypeSymbol with the given underlying type and (optional) element names.
         /// </summary>
         /// <remarks>
         /// Since VB doesn't support tuples yet, this call will fail in a VB compilation.
+        /// Also, the underlying type needs to be tuple-compatible.
         /// </remarks>
-        public INamedTypeSymbol CreateTupleTypeSymbol(ImmutableArray<ITypeSymbol> elementTypes, ImmutableArray<string> elementNames = default(ImmutableArray<string>))
+        public INamedTypeSymbol CreateTupleTypeSymbol(INamedTypeSymbol underlyingType, ImmutableArray<string> elementNames = default(ImmutableArray<string>))
```

Comment:
I think this is an ill-advised change to the public API. We shouldn't want the client to have to construct the nested instantiations of ValueTuple in order to properly "create" a tuple type. The details of that nesting are best abstracted by the compiler's public API as much as possible.